lib/web_ui/lib/src/engine/window.dart

```diff
@@ -689,6 +689,46 @@ class EngineWindow extends ui.Window {
     // callback(null);
   }
 
+  bool _isHistoryInitialized = false;
+
+  @visibleForTesting
+  Future<void> debugResetHistory() async {
+    _isHistoryInitialized = false;
+    urlStrategy = null;
+    await _browserHistory.setUrlStrategy(null);
+    _browserHistory = MultiEntriesBrowserHistory();
+  }
+
+  Future<bool> _handleNavigationMessage(
+    ByteData? data,
+    ui.PlatformMessageResponseCallback? callback,
+  ) async {
+    const MethodCodec codec = JSONMethodCodec();
+    final MethodCall decoded = codec.decodeMethodCall(data);
+    final Map<String, dynamic> arguments = decoded.arguments;
+
+    if (!_isHistoryInitialized) {
+      await setUrlStrategy(urlStrategy);
```

Comment:
Having `get urlStrategy`, `set urlStrategy`, `setUrlStrategy`, and `_browserHistory.setUrlStrategy` is confusing. I think removing the need for conversion and simplifying the `get urlStrategy` can fix this (see other comments).